selfhost/typechecker.jakt

```diff
@@ -4641,12 +4661,73 @@ struct Typechecker {
     }
 
     function typecheck_if(mut this, condition: ParsedExpression, then_block: ParsedBlock, else_statement: ParsedStatement?, scope_id: ScopeId, safety_mode: SafetyMode, span: Span) throws -> CheckedStatement {
+        let can_throw = .get_scope(scope_id).can_throw
+
+        match condition {
+            BinaryOp(lhs, op, rhs) => {
+                if op is LogicalAnd {
+                    match lhs {
+                        UnaryOp(expr, op: inner_op) => {
```

Comment:
Can we do this recursively instead?

Something like this should also work:

```
if x is Foo(a) and y is Bar(b) and a < b { ... }
```

Reply:
Will add a third test with this case :)

Reply:
We chatted and the idea is to typecheck the condition separately so we can reuse the function (for example, with guard in the future)

This implemention also did yet handle the case where the condition starts with something that isn't `is`

For example: `if 3 < 4 and foo is Foo(b) { ... }`